src/google/adk_community/artifacts/s3_artifact_service.py

```diff
@@ -0,0 +1,694 @@
+  ) -> list[int]:
+    return await asyncio.to_thread(
+        self._list_versions_sync,
+        app_name,
+        user_id,
+        session_id,
+        filename,
+    )
+
+  def _save_artifact_sync(
+      self,
+      app_name: str,
+      user_id: str,
+      session_id: Optional[str],
+      filename: str,
+      artifact: types.Part,
+      custom_metadata: Optional[dict[str, Any]],
+  ) -> int:
+    """Synchronous implementation of save_artifact."""
+    # Get next version number
+    versions = self._list_versions_sync(
+        app_name=app_name,
+        user_id=user_id,
+        session_id=session_id,
+        filename=filename,
+    )
+    version = 0 if not versions else max(versions) + 1
+
+    object_key = self._get_object_key(
+        app_name, user_id, filename, version, session_id
+    )
+
+    # Prepare data and content type
+    if artifact.inline_data:
+      data = artifact.inline_data.data
+      content_type = artifact.inline_data.mime_type or "application/octet-stream"
+    elif artifact.text:
+      data = artifact.text.encode("utf-8")
+      content_type = "text/plain; charset=utf-8"
+    else:
+      raise InputValidationError(
+          "Artifact must have either inline_data or text content."
+      )
+
+    # Prepare put_object arguments
+    put_kwargs: dict[str, Any] = {
+        "Bucket": self.bucket_name,
+        "Key": object_key,
+        "Body": data,
+        "ContentType": content_type,
+    }
+
+    # Add custom metadata if provided
+    if custom_metadata:
+      put_kwargs["Metadata"] = {
+          str(k): str(v) for k, v in custom_metadata.items()
+      }
+
+    try:
+      self.s3_client.put_object(**put_kwargs)
+      logger.debug(
+          "Saved artifact %s version %d to S3 key %s",
+          filename,
+          version,
+          object_key,
+      )
+      return version
+    except ClientError as e:
+      error_code = e.response.get("Error", {}).get("Code", "Unknown")
+      logger.error(
+          "Failed to save artifact '%s' to S3: %s (Error: %s)",
+          filename,
+          e,
+          error_code,
+      )
+      raise
+    except Exception as e:
+      logger.error(
+          "Unexpected error saving artifact '%s' to S3: %s", filename, e
+      )
+      raise
+
+  def _load_artifact_sync(
+      self,
+      app_name: str,
+      user_id: str,
+      session_id: Optional[str],
+      filename: str,
+      version: Optional[int],
+  ) -> Optional[types.Part]:
+    """Synchronous implementation of load_artifact."""
+    if version is None:
+      versions = self._list_versions_sync(
+          app_name=app_name,
+          user_id=user_id,
+          session_id=session_id,
+          filename=filename,
+      )
+      if not versions:
+        return None
+      version = max(versions)
+
+    object_key = self._get_object_key(
+        app_name, user_id, filename, version, session_id
+    )
+
+    try:
+      response = self.s3_client.get_object(
```

Comment:
Is there any async version of method in s3_client?  If so, can we use them directly so we don't need to use asyncio?